### PR TITLE
Switch status alerts from colored text to colored icons

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

In the new desktop theme, our accent colors for status alerts (success/warning/danger) only meet the [3:1 contrast ratio minimum](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast), which is only permitted for icons/graphical elements, not text. I did it this way in order to have a bright yellow/orange color for warnings that isn't super dark and weird looking. Thus, we can't have text with these colors.

I think this is a fine constraint, because we rarely use colored text, and it's still effective to have a colored icon to grab attention accompanied by text with the default text color.

To implement this new approach, this PR:
- Adds a `color` prop to the `Icon` components
- Removes the `color` prop from the `Font` components
- Updates all the status message use cases accordingly

_Review by commit_

## Demo Video or Screenshot
A few sample changes from VxMark
Before
Screenshot 2023-11-01 at 10 12 17 AM
After
Screenshot 2023-11-01 at 10 21 55 AM

Before
Screenshot 2023-11-01 at 10 13 48 AM
After
Screenshot 2023-11-01 at 10 21 09 AM

## Testing Plan
- Automated tests
- Visual inspection of most of the changes to make sure they are still legible

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/frontend/src/components/import_cvrfiles_modal.tsx [hunk 1]
```diff
@@ -394,13 +395,13 @@ export function ImportCvrFilesModal({ onClose }: Props): JSX.Element | null {
...
-                  <Font color="success">Official</Font>
-                )}
-              </LabelText>
+              {isTestModeResults ? (
+                <LabelText>
+                  <Icons.Warning color="warning" /> Test
+                </LabelText>
+              ) : (
+                <LabelText>Official</LabelText>
+              )}
```
Comment (jonahkagan) [inline apps/admin/frontend/src/components/import_cvrfiles_modal.tsx:404]: Screenshot 2023-11-01 at 10 28 58 AM

File: apps/admin/frontend/src/components/smartcard_modal/status_message.tsx [hunk 2]
```diff
@@ -103,16 +101,18 @@ export function SuccessOrErrorStatusMessage({
   if (action === 'PinReset') {
     assert(newPin !== undefined);
     return (
-      <TextLarge color="success">
-        New card PIN is <Font weight="bold">{hyphenatePin(newPin)}</Font>.
+      <TextLarge>
+        <Icons.Done color="success" /> New card PIN is{' '}
+        <Font weight="bold">{hyphenatePin(newPin)}</Font>.
```
Comment (jonahkagan) [inline apps/admin/frontend/src/components/smartcard_modal/status_message.tsx:106]: Screenshot 2023-11-01 at 10 29 50 AM

**Review by adghayes — APPROVED** [review]
Looks better this way!

Is the "warning" red instead of orange just to draw attention / increase contrast? I'm very used to the warnings being orange, and now the `warning` vs. `danger` colors are exactly the same for icons.

**Comment by jonahkagan** [comment]
> Is the "warning" red instead of orange just to draw attention / increase contrast? I'm very used to the warnings being orange, and now the `warning` vs. `danger` colors are exactly the same for icons.

Oh, I got confused. I thought it was already red on `main`, but it turns out I made that change in an earlier PR (https://github.com/votingworks/vxsuite/commit/cdde8fcd5cd681a9696ebdce209bf6ce4b47ac46#diff-01f30c555c79c11331b594d7ca37e59459b212f73b76186e19e83693d080776cR247). I'm going to switch it back to orange, which I agree is the correct choice.

**Pull request merged**